adding support for net8 assembly list in emg

brettsam opened the pull request:
<!-- Please provide all the information below.  -->

resolves #10112

Adding support for separate assembly cleanup lists for net6 and net8 in ExtensionsMetadataGenerator.

### Pull request checklist

**IMPORTANT**: Currently, changes must be backported to the `in-proc` branch to be included in Core Tools and non-Flex deployments.

* [x] Backporting to the `in-proc` branch is not required
    * Otherwise: Link to backporting PR
* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] My changes **do not** require diagnostic events changes
    * Otherwise: I have added/updated all related diagnostic events and their documentation (Documentation issue linked to PR)
* [x] I have added all required tests (Unit tests, E2E tests)

brettsam committed:
adding support for net8 assembly list in emg

fabiocav commented on tools/ExtensionsMetadataGenerator/src/ExtensionsMetadataGenerator/BuildTasks/RemoveRuntimeDependencies.cs:
```diff
@@ -32,8 +38,14 @@ public override bool Execute()
 
+            if (!_runtimeToAssemblyFileMap.TryGetValue(TargetFramework, out var assemblyListFileName))
+            {
+                Log.LogError($"The TargetFramework '{TargetFramework}' is not supported in this project.");
```
Wondering if we could run into problems here. We might still have users targeting different TFMs other than net6.0 (e.g. .NET standard). We could still enforce this here, but wondering if we should improve the message to list the supported targets. Thoughts?

fabiocav commented on tools/ExtensionsMetadataGenerator/src/ExtensionsMetadataGenerator/ExtensionsMetadataGenerator.csproj:
```diff
@@ -37,11 +37,11 @@
-      <Content Include="$(OutputPath)\netstandard2.0\generator\*">
-        <Pack>true</Pack>
+      <Content Include="$(ArtifactsPath)\bin\ExtensionsMetadataGenerator\release_netstandard2.0\generator\*">
+        <Pack>true</Pack>        
```
nit: if making other changes; space.

liliankasem commented:
I separated the EMG tests from the rest of the unit tests because of multi-targeting issues; can you group them back together and see if things work properly now?

https://github.com/Azure/azure-functions-host/blob/cea988b1753e2de4ec64b35cb55f31203d453b8e/azure-pipelines.yml#L322-L329

jviau commented:
I am curious - how much work would it be to rewrite this as an .netstandard2.0 msbuild task?

brettsam commented:
> I am curious - how much work would it be to rewrite this as an .netstandard2.0 msbuild task?

@jviau -- Which part? The ExtensionsMetadataGenerator assembly (the one with the Tasks) currently multi-targets netstandard2.0 and net46. I think we had to do that because of how VS loads the tasks and we needed the AppDomainIsolatedTask for that... @fabiocav may remember the exact details.

brettsam committed:
re-adding emg tests

brettsam commented:
> I separated the EMG tests from the rest of the unit tests because of multi-targeting issues; can you group them back together and see if things work properly now?

@liliankasem -- I reverted this, added them back, saw the build continue to fail. Then I realized... oh wait, splitting this out is actually the right approach. We'll always build this solution as net6 and it's unrelated to the host framework. So, I'm keeping this split out.

brettsam committed:
breaking emg tests back out

liliankasem commented:
> > I separated the EMG tests from the rest of the unit tests because of multi-targeting issues; can you group them back together and see if things work properly now?
> 
> @liliankasem -- I reverted this, added them back, saw the build continue to fail. Then I realized... oh wait, splitting this out is actually the right approach. We'll always build this solution as net6 and it's unrelated to the host framework. So, I'm keeping this split out.

Thank you for checking!

brettsam committed:
adding earlier frameworks for mapping

brettsam committed:
tweaks
